pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/ClusterData.java

```diff
@@ -58,6 +60,20 @@
         example = "pulsar+ssl://pulsar.example.com:6651"
     )
     private String brokerServiceUrlTls;
+    @ApiModelProperty(
+        name = "proxyServiceUrl",
+        value = "Proxy-service url when client would like to connect to broker via proxy.",
+        example = "pulsar+ssl://ats-proxy.example.com:4443 or "
+                    + "pulsar://ats-proxy.example.com:4080"
+    )
+    private String proxyServiceUrl;
```

Comment:
based on my understanding, this service URL here can only be the binary protocol service URL correct? We can't use the HTTP-protocol service URL. If that's the case, can you document this setting in `ClusterData` and the related command-line tools? It might be good to add a validation to throw exceptions if it is not a binary protocol service url.

Reply:
yes, SNI works on layer-4 so, it requires binary URL. let me add such documentation. Also, I have a plan to add a separate document with examples of different proxy types as multiple users are interested in this solution.